libcudacxx/include/cuda/std/__exception/cuda_error.h

```diff
@@ -27,41 +27,59 @@
 #endif // !_CCCL_CUDA_COMPILER_NVCC && !_CCCL_CUDA_COMPILER_NVHPC
 
 #include <cuda/std/__exception/terminate.h>
-#include <cuda/std/detail/libcxx/include/stdexcept>
 
-#include <nv/target>
 #if !defined(_CCCL_COMPILER_NVRTC)
 #  include <cstdio>
+#  include <stdexcept>
 #endif // !_CCCL_COMPILER_NVRTC
 
+#include <nv/target>
+
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
 /**
  * @brief Exception thrown when a CUDA error is encountered.
  */
-class cuda_error : public _CUDA_VSTD_NOVERSION::runtime_error
+
+#ifndef _LIBCUDACXX_NO_EXCEPTIONS
+class cuda_error : public ::std::runtime_error
 {
-public:
-  cuda_error(::cudaError_t __status, const char* __msg) noexcept
-      : _CUDA_VSTD_NOVERSION::runtime_error("")
+private:
+  struct __msg_storage
+  {
+    char __buffer[256];
+  };
+
+  static char* __format_string(::cudaError_t __status, const char* __msg, char __formatted_msg[256]) noexcept
```

Comment:
Suggestion: Maybe we should prefer spelling a pointer, which is what a built-in array parameter is anyway, for clarity.
```suggestion
  static char* __format_string(::cudaError_t __status, const char* __msg, char* __formatted_msg) noexcept
```